test.js

```diff
@@ -0,0 +1,124 @@
+const app = require('express')()
+const bodyParser = require('body-parser')
+const multer = require('multer') // v1.0.5
+const upload = multer() // for parsing multipart/form-data
+
+app.use(bodyParser.json()) // for parsing application/json
+app.use(bodyParser.urlencoded({ extended: true })) // for parsing application/x-www-form-urlencoded
+
+app.post('/profile', upload.array(), function (req, res, next) {
+  console.log(req.body)
+  res.json(req.body)
+})
+
+const port = 3001;
+
+app.get('/', (req, res) => res.send('Hello World!'));
+
+
+class Treenode {
+
+  constructor(label) {
+    this.id = (++Treenode.id_gen).toString()
+    // Treenode.lineage[this.id]=[label]
+    this.label =  label
+    this.parent_id = null
+    // this.parent_label = null
+    this.children = []
+    // Treenode.label_id[this.id] = label
+  }
+
+  add_child(child){
+    this.children.push(child)
+    child.parent_id = this.id
+  }
+
+}
+
+function find_id(iter, parent_id) {
+  if (iter.id === parent_id) {
+    return iter
+  }
+  else {
+    if (iter.children.length > 0) {
+      for (var i = 0; i < iter.children.length; i++){
+        retVal = find_id(iter.children[i], parent_id)
+        if (retVal !== null) {
+          return retVal
+        }
+      }
+    }
+  }
+  return null
+}
+
+Treenode.id_gen = 0;
+Treenode.label_id = {};
+
+root = new Treenode('root')
+ant = new Treenode('ant')
+bear = new Treenode('bear')
+cat = new Treenode('cat')
+dog = new Treenode('dog')
+elephant = new Treenode('elephant')
+frog = new Treenode('frog')
+
+root.add_child(ant)
+root.add_child(bear)
+bear.add_child(cat)
+bear.add_child(dog)
+dog.add_child(elephant)
+root.add_child(frog)
+
+
+var str = JSON.stringify(root, null, 5);
+
+app.get('/api/tree', function (req, res) {
+  res.send(root);
+});
+
+app.post('/api/tree', function (req, res) {
+  // {"parent": "<id>", "label":<label>}
+  var child = new Treenode(req.body.label);
+  var parent_id = req.body.parent.toString();
+  var parent_node = Treenode.label_id[parent_id]
+
+  parent = find_id(root, parent_id)
+  if (parent !== null) {
+    parent.add_child(child)
+  }
+  else{ label = new Treenode(label)}
+  res.send(root)
+
+});
+
+app.delete('/api/tree/:id', function (req,res) {
```

Comment:
This comment goes for all your apis
The main goal of these routes and functions is to receive the request, deserialize it, execute business logic, send response.

this method does all four. Ideally it only handles the receiving request, and sending the response. JS automatically deserializes the json request.body so no problem there.
If we want this to be better we could refactor the business logic to the manager or outside this, and call that method, then send the correct response based on what we get from that `deleteNode` method